$OpenBSD$

Add a compile-time option to make Tor browser look for its data
directory in the user's home directory and turn it on for OpenBSD
--- xpcom/io/TorFileUtils.cpp.orig	Mon Sep  5 19:58:54 2016
+++ xpcom/io/TorFileUtils.cpp	Mon Sep  5 19:58:59 2016
@@ -6,6 +6,10 @@
 
 #include "TorFileUtils.h"
 
+#if defined(__OpenBSD__)
+# define TOR_BROWSER_DATA_IN_HOME_DIR 1
+#endif /* __OpenBSD__ */
+
 static nsresult GetAppRootDir(nsIFile *aExeFile, nsIFile** aFile);
 
 //-----------------------------------------------------------------------------
@@ -14,9 +18,18 @@ TorBrowser_GetUserDataDir(nsIFile *aExeFile, nsIFile**
 {
   NS_ENSURE_ARG_POINTER(aFile);
   nsCOMPtr<nsIFile> tbDataDir;
-
-#ifdef TOR_BROWSER_DATA_OUTSIDE_APP_DIR
   nsAutoCString tbDataLeafName(NS_LITERAL_CSTRING("TorBrowser-Data"));
+
+#if defined(TOR_BROWSER_DATA_IN_HOME_DIR)
+  const char *homeDir = getenv("HOME");
+  if (!homeDir || !*homeDir)
+    return NS_ERROR_FAILURE;
+  nsresult rv = NS_NewNativeLocalFile(nsDependentCString(homeDir), true,
+                                      getter_AddRefs(tbDataDir));
+  NS_ENSURE_SUCCESS(rv, rv);
+  rv = tbDataDir->AppendNative(tbDataLeafName);
+  NS_ENSURE_SUCCESS(rv, rv);
+#elif defined(TOR_BROWSER_DATA_OUTSIDE_APP_DIR)
   nsCOMPtr<nsIFile> appRootDir;
   nsresult rv = GetAppRootDir(aExeFile, getter_AddRefs(appRootDir));
   NS_ENSURE_SUCCESS(rv, rv);
